smithy-codegen-core/src/test/java/software/amazon/smithy/codegen/core/DefinitionsTest.java

```diff
@@ -0,0 +1,125 @@
+package software.amazon.smithy.codegen.core;
+
+import org.junit.jupiter.api.Test;
+import software.amazon.smithy.model.node.ObjectNode;
+
+import java.io.FileNotFoundException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+class DefinitionsTest {
+
+    @Test
+    void toNode() {
+        Definitions definitions1 = new Definitions();
+
+        definitions1.setTags(new HashMap<>());
+        definitions1.setTypes(new HashMap<>());
+
+        Map<String, String> typesMap = definitions1.getTypes();
+        Map<String, String> tagsMap = definitions1.getTags();
+
+        typesMap.put("t1", "t1val");
+        typesMap.put("t2", "t2val");
+        tagsMap.put("tag1", "tag1val");
+        tagsMap.put("tag2", "tag2val");
+
+        ObjectNode node = definitions1.toNode();
+
+        assert node.containsMember(Definitions.TYPE_TEXT);
```

Comment:
This can be cleaned up using [the `Node.assertEquals` method](https://github.com/awslabs/smithy/blob/bbb166dabafd2da7bfe18c1fbd1e1713ec96a2f9/smithy-model/src/main/java/software/amazon/smithy/model/node/Node.java#L314).